src/NuGet.Core/NuGet.Protocol/Resources/PackageMetadataResourceV3.cs

```diff
@@ -121,9 +121,10 @@ public class PackageMetadataResourceV3 : PackageMetadataResource
                         var rangeUri = registrationPage.Url;
                         var leafRegistrationPage = await GetRegistratioIndexPageAsync(_client, rangeUri, packageId, lower, upper, httpSourceCacheContext, log, token);
 
-                        if (registrationPage == null)
+                        if (leafRegistrationPage == null)
                         {
-                            throw new InvalidDataException(registrationUri.AbsoluteUri);
+                            log.LogDebug($"{rangeUri} returned 404 NotFound. No versions in this range are available.");
```

Comment:
No not sure how it behaved before that. It 404's because all the versions within the leaf are have been unlisted. I can't comment on why cloudsmith opts to 404 when there are no active versions in a range but just based on the `IgnoreNotFound` flag that's being set here, I feel like this is a valid way of handling this maybe?

Reply:
I can force cloudsmith to synchronize at least one package from the range that 404's by explicitly installing an older version. That fixes those urls so they don't 404 for me but I feel like this behavior change between .net sdk 6 and .net sdk 8 is not desirable. Whether that's NuGet's responsibility I'm not sure but this seemed like a reasonable implementation to me.

Reply:
>  It 404's because all the versions within the leaf have been unlisted

I think there's a couple of things that kind of seem incorrect. 

- Registration should not be filtering unlisted packages. 
- If all packages from a package don't "exist", then the page should not exist either. 

In general, I think this change won't really lead to any issues with current servers, since most servers won't have empty pages, but it still feels like this should be a server side fix.

Reply:
We tried but couldn't find any way to disable this behavior on the Cloudsmith side. Since we have a workaround in place and this isn't a blocker for us I'm happy to change this PR to just check the correct variable if that's what we want. But I still think this change makes sense in terms of defensive programming.

Reply:
I think the check on line 124 makes sense. Currently it's checking the wrong value. 

I'm not sure I'd call continue instead of throwing defensive programming. There's a client/server contract and the server is not honoring it. Throwing is a controlled failure.

Reply:
If we're going to consider this a data contract error would it make sense to get rid of the null check altogether and set `IgnoreNotFounds` to `false` ?
https://github.com/NuGet/NuGet.Client/blob/3088606e8ab4b499adb8b645d056029411826e6d/src/NuGet.Core/NuGet.Protocol/Resources/PackageMetadataResourceV3.cs#L199

Not sure why we're handling 404 differently in this case. We can treat it like any other http error code.

Reply:
@nkolev92 Let me know what you think of my latest changes.

I set `IgnoreNotFounds` to false and adjusted my unit test accordingly. I also removed the null check since we [ensure a successful http status code](https://github.com/NuGet/NuGet.Client/blob/7ad6fcc9c56c960975c37b2416c7eae1d53ba3fd/src/NuGet.Core/NuGet.Protocol/HttpSource/HttpSource.cs#L155) in the http handler.

Reply:
@nkolev92 Any thoughts on the latest version of the code?